src/Dialect/ONNX/ONNXOps/Quantize/DequantizeLinear.cpp

```diff
@@ -127,9 +127,11 @@ LogicalResult ONNXDequantizeLinearOp::verify() {
     //   return emitOpError("x and x_zero_point must have the same data type");
 
     if (getElementType(zero.getType()).isInteger(32) && zeroLen != 0)
-      if (auto values = getElementAttributeFromONNXValue(zero))
-        if (!values.isSplat() || !values.getSplatValue<APInt>().isZero())
+      if (auto values = getElementAttributeFromONNXValue(zero)) {
+        WideNum zero = WideNum::widen<BType::INT32>(0);
```

Comment:
I am not sure why x_zero_point should be compared to constant 0. I did not see that requirement in the specification.
Another comment is about allEqual: if the element type of 'values' is not INT32, will allEqual return false?

Reply:
> I am not sure why x_zero_point should be compared to constant 0. I did not see that requirement in the specification.

The spec says "In the case of dequantizing int32, there's no zero point (zero point is supposed to be 0)"
https://github.com/onnx/onnx/blob/main/docs/Operators.md#DequantizeLinear

> Another comment is about allEqual: if the element type of 'values' is not INT32, will allEqual return false?

WideNum is an untagged union so allEqual(values, zero) assumes zero has the same type as values' element type, it cannot check that it does. Here the if condition ensures values has element type i32 and correspondingly we encode zero as INT32.